pkg/api/crush_rule_api_handlers.go

```diff
@@ -0,0 +1,139 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+
+	pb "github.com/seaphony/ceph-api/api/gen/grpc/go"
+	"github.com/seaphony/ceph-api/pkg/rados"
+	"github.com/seaphony/ceph-api/pkg/types"
+	"github.com/seaphony/ceph-api/pkg/user"
+
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func NewCrushRuleAPI(radosSvc *rados.Svc) pb.CrushRuleServer {
+	return &crushRuleAPI{
+		radosSvc: radosSvc,
+	}
+}
+
+type crushRuleAPI struct {
+	radosSvc *rados.Svc
+}
+
+type crushDump struct {
+	Rules []*pb.Rule `json:"rules"`
+}
+
+func (c *crushRuleAPI) CreateRule(ctx context.Context, req *pb.CreateRuleRequest) (*emptypb.Empty, error) {
+	if err := user.HasPermissions(ctx, user.ScopeOsd, user.PermCreate); err != nil {
+		return nil, err
+	}
+
+	// check if name and failure domain are set
+	if req.Name == "" || req.FailureDomain == "" {
+		return nil, types.ErrInvalidArg
+	}
```

Comment:
Please add info for validation errors:
```suggestion
	if req.Name == "" {
		return nil, fmt.Errorf("%w: name is required", types.ErrInvalidArg)
	}
	if req.FailureDomain == "" {
		return nil, fmt.Errorf("%w: failure domain is required", types.ErrInvalidArg)
	}
```